components/user_settings/advanced/user_settings_advanced.tsx

```diff
@@ -1,57 +1,83 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 /* eslint-disable react/no-string-refs */
+/* eslint-disable max-lines */
 
-import PropTypes from 'prop-types';
-import React from 'react';
+import React, {ReactNode} from 'react';
 import {FormattedMessage} from 'react-intl';
 
 import {emitUserLoggedOutEvent} from 'actions/global_actions';
+
 import Constants, {Preferences} from 'utils/constants';
-import * as Utils from 'utils/utils';
 import {t} from 'utils/i18n';
+import {isMac, localizeMessage} from 'utils/utils';
+
 import SettingItemMax from 'components/setting_item_max.jsx';
 import SettingItemMin from 'components/setting_item_min';
 import ConfirmModal from 'components/confirm_modal';
 import BackIcon from 'components/widgets/icons/fa_back_icon';
 
+import {UserProfile} from '@mattermost/types/users';
+import {PreferenceType} from '@mattermost/types/preferences';
+
+import {ActionResult} from 'mattermost-redux/types/actions';
+
 import JoinLeaveSection from './join_leave_section';
 import PerformanceDebuggingSection from './performance_debugging_section';
 
 const PreReleaseFeatures = Constants.PRE_RELEASE_FEATURES;
 
-export default class AdvancedSettingsDisplay extends React.PureComponent {
-    static propTypes = {
-        currentUser: PropTypes.object.isRequired,
-        advancedSettingsCategory: PropTypes.array.isRequired,
-        isAdvancedTextEditorEnabled: PropTypes.bool,
-        sendOnCtrlEnter: PropTypes.string.isRequired,
-        codeBlockOnCtrlEnter: PropTypes.bool,
-        formatting: PropTypes.string.isRequired,
-        joinLeave: PropTypes.string.isRequired,
-        unreadScrollPosition: PropTypes.string.isRequired,
-        updateSection: PropTypes.func,
-        activeSection: PropTypes.string,
-        closeModal: PropTypes.func.isRequired,
-        collapseModal: PropTypes.func.isRequired,
-        enablePreviewFeatures: PropTypes.bool,
-        enableUserDeactivation: PropTypes.bool,
-        actions: PropTypes.shape({
-            savePreferences: PropTypes.func.isRequired,
-            updateUserActive: PropTypes.func.isRequired,
-            revokeAllSessionsForUser: PropTypes.func.isRequired,
-        }).isRequired,
-    }
+type Setting = {
+    [key: string]: string | undefined;
```

Comment:
question: do we really have generic options we need to take into account?

Reply:
Sorry, I didn't get the question 🥲

Reply:
Thanks for the explanation. Now I see your point. 

The issue is that `as.name` can change in the future

https://github.com/mattermost/mattermost-webapp/blob/bd3fafaf1273fcf1ef593eb2a528545581eac3a1/components/user_settings/advanced/user_settings_advanced.tsx#L99-L100

Currently there is only one option (i.e. `feature_enabled_ markdown_preview` coming from below. If there will be more `PRE_RELEASE_FEATURES` in the future, I suppose it might create type issues

https://github.com/mattermost/mattermost-webapp/blob/bd3fafaf1273fcf1ef593eb2a528545581eac3a1/utils/constants.tsx#L1725-L1731